### PR TITLE
test: join automation updated

### PR DESCRIPTION
**Pull request by ShyamOOAI**
## Summary by CodeRabbit

- **New Features**
	- Added a quick mode toggle button on the Logs Page.
	- Introduced new methods for validating log displays and error messages.
	- New test cases to verify the functionality of toggling quick mode and interacting with interesting fields.

- **Bug Fixes**
	- Enhanced test scenarios to ensure correct display of streams and count queries during log searches.
	- Improved verification of results after executing queries.

### CODE REVIEW

File: tests/ui-testing/pages/logsPage.js [hunk 1]
```diff
@@ -285,10 +285,19 @@ async logsPageURLValidation() {
     await this.page.waitForTimeout(5000);
   }
 
+  async displayCountQuery() {
+    await this.page.waitForTimeout(4000);
+    await expect(this.page.getByRole('heading', { name: 'Error while fetching' })).not.toBeVisible();
+  }
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/pages/logsPage.js:291]: @ShyamOOAI - please check this as well and rename the method

[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:299]: have removed

File: tests/ui-testing/pages/logsPage.js [hunk 2]
```diff
@@ -285,10 +285,19 @@
...
+  
 
   async signOut() {
     await this.profileButton.click();
     await this.signOutButton.click();
   }
 
+  async displayTwoStreams() {
+    await expect(this.page.locator('[data-test="logs-search-index-list"]')).toContainText('default, e2e_automate');
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:319]: simple to understand better

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 3]
```diff
@@ -87,6 +93,7 @@
         await logsPage.kubernetesContainerNameLeftJoin();
         await logsPage.enableSQLMode();
         await logsPage.selectRunQuery();
+        await logsPage.displayCountQuery();
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js]: Have Added

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 4]
```diff
@@ -43,6 +46,7 @@ test.describe("Join for logs", () => {
         await logsPage.kubernetesContainerName();
         await logsPage.enableSQLMode();
         await logsPage.selectRunQuery();
+        await logsPage.displayCountQuery();
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js]: have updated

File: tests/ui-testing/pages/logsPage.js [hunk 5]
```diff
@@ -285,10 +293,30 @@ async logsPageURLValidation() {
...
+
+  async verifyQuickModeToggle() {
+    await expect(this.page.locator(this.quickModeToggle)).toBeVisible();
+  }
+
+  async verifyQuickModeToggle() {
+
+    await expect(this.page.locator(this.quickModeToggle)).toBeVisible();
+    
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js]: have updated

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 6]
```diff
@@ -12,6 +13,7 @@
         ingestionPage = new IngestionPage(page);
         logsPage = new LogsPage(page);
         await loginPage.gotoLoginPage();
+        await loginPage.loginAsInternalUser();
         await loginPage.login(); // Login as root user
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js:17]: updated

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 7]
```diff
@@ -21,6 +21,7 @@ test.describe("Join for logs", () => {
 
         await logsPage.navigateToLogs();
         await logsPage.selectIndexAndStreamJoin();
+        await logsPage.displayTwoStreams();
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js:26]: As per required steps

File: tests/ui-testing/pages/logsPage.js [hunk 8]
```diff
@@ -217,6 +218,13 @@
     await expect.poll(async () => (await search).status()).toBe(200);
   }
 
+  async applyQueryButton(expectedUrl) {
+    const searchResponse = this.page.waitForResponse(expectedUrl);
+    await this.page.waitForTimeout(3000);
+    await this.page.locator(this.queryButton).click({ force: true });
+    await expect.poll(async () => (await searchResponse).status()).toBe(200);
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:226]: have updated

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 9]
```diff
@@ -109,7 +119,22 @@ test.describe("Join for logs", () => {
...
+    test("Display quick mode toggle button", async ({ page }) => {
+
+        await logsPage.navigateToLogs();
+        await logsPage.selectIndexAndStreamJoin();
+        await logsPage.enableSQLMode();
+        await logsPage.selectRunQuery();
+        await logsPage.verifyQuickModeToggle();
+ 
+
+    });
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js]: ok

File: tests/ui-testing/pages/logsPage.js [hunk 10]
```diff
@@ -285,10 +293,40 @@ async logsPageURLValidation() {
...
+    await expect(this.page.getByRole('heading', { name: 'Error while fetching' })).not.toBeVisible();
+  }
+  
+  async validateResult() {
+    await this.page.waitForTimeout(10000);
+    await expect(this.page.locator('[data-test="logs-search-search-result"]')).toBeVisible();
+    await this.page.locator('td[data-test="log-table-column-0-source"]').nth(0).click();
+    await expect(this.page.locator('[data-test="log-expand-detail-key-kubernetes_container_name-text"]')).toBeVisible();
+   
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js]: Similar to understand

File: tests/ui-testing/pages/logsPage.js [hunk 11]
```diff
@@ -235,60 +250,132 @@
...
+
+    await this.page.locator('[data-cy="index-field-search-input"]').clear();
+    await this.page.locator('[data-cy="index-field-search-input"]').fill("job");
+    await this.page.waitForTimeout(2000);
+    await this.page.locator('[data-test="log-search-index-list-interesting-job-field-btn"]').last().click({force: true,});
+    await this.page.locator('[data-cy="search-bar-refresh-button"] > .q-btn__content').click({force: true,});
+    await this.page.locator('[data-test="log-search-index-list-interesting-job-field-btn"]').last().click({force: true, });
+    await expect(this.page.locator('[data-test="logs-search-bar-query-editor"]')).not.toHaveText(/job/);
+   
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:375]: next

File: tests/ui-testing/pages/logsPage.js [hunk 12]
```diff
@@ -214,7 +215,21 @@ async logsPageURLValidation() {
...
-    await expect.poll(async () => (await search).status()).toBe(200);
+    try {
+      const response = await searchResponse;
+      await expect.poll(async () => response.status()).toBe(200);
+      
+        } 
+      catch (error) {
+       throw new Error(`Failed to get response from ${expectedUrl}: ${error.message}`);
+       }   
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:226]: have updated

File: tests/ui-testing/playwright-tests/join.spec.js [hunk 13]
```diff
@@ -109,7 +120,43 @@ test.describe("Join for logs", () => {
...
+    test("Click on interesting field icon and display field in editor", async ({ page }) => {
+
+        await logsPage.navigateToLogs();
+        await logsPage.selectIndexAndStreamJoin();
+        await logsPage.enableSQLMode();
+        await logsPage.selectRunQuery();
+        await logsPage.clickQuickModeToggle();
+        await logsPage.clickInterestingFields();
+        await logsPage.validateInterestingFields();
+    });
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/join.spec.js:137]: easy to understand

File: tests/ui-testing/pages/logsPage.js [hunk 14]
```diff
@@ -235,60 +243,132 @@
...
+
+    await this.page.locator('[data-cy="index-field-search-input"]').clear();
+    await this.page.locator('[data-cy="index-field-search-input"]').fill("job");
+    await this.page.waitForTimeout(2000);
+    await this.page.locator('[data-test="log-search-index-list-interesting-job-field-btn"]').last().click({force: true,});
+    await this.page.locator('[data-cy="search-bar-refresh-button"] > .q-btn__content').click({force: true,});
+    await this.page.locator('[data-test="log-search-index-list-interesting-job-field-btn"]').last().click({force: true, });
+    await expect(this.page.locator('[data-test="logs-search-bar-query-editor"]')).not.toHaveText(/job/);
+   
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:375]: next time

File: tests/ui-testing/pages/logsPage.js [hunk 15]
```diff
@@ -235,60 +243,132 @@ async logsPageURLValidation() {
...
+    const toggleButton = await this.page.$('[data-test="logs-search-bar-quick-mode-toggle-btn"] > .q-toggle__inner');
+    // Evaluate the class attribute to determine if the toggle is in the off state
+    const isSwitchedOff = await toggleButton.evaluate(node => node.classList.contains('q-toggle__inner--falsy'));
+    // If the toggle is switched off, click on it to switch it on
+    if (isSwitchedOff) {
+      await toggleButton.click();
+
+    }
+
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:345]: As per suggestion

**Review by neha00290 — APPROVED** [review]

**Pull request merged**